scripts/update-browser-releases/safari.ts

```diff
@@ -82,117 +81,65 @@ export const updateSafariReleases = async (options) => {
-          safariBCD.browsers[options.bcdBrowserName].releases
-        )
-      ) {
-        if (Number(releaseData.version) > 15) {
-          // We know that version past Safari 15 matches iOS versions too
-          console.warn(
-            chalk`{yellow Old version ${releaseData.version} not found in BCD file}`,
-          );
-        }
-        continue;
-      }
-      const engineStored =
-        safariBCD.browsers[options.bcdBrowserName].releases[releaseData.version]
-          .engine_version;
-      if (releaseData.engine !== engineStored) {
-        // Differs!
-        console.warn(
-          chalk`{yellow Engine for ${releaseData.version} (${releaseData.engine}) doesn't match engine stored (${engineStored})}`,
-        );
-      }
+    // Don't use the date for beta, we only record release dates, not beta dates
+    if (releaseDataEntry.channel === 'beta') {
+      releaseDataEntry.date = '';
     }
-  }
 
-  //
-  // Update stable release
-  //
-  if (
-    safariBCD.browsers[options.bcdBrowserName].releases[stableRelease.version]
-  ) {
-    result += updateBrowserEntry(
-      safariBCD,
-      options.bcdBrowserName,
-      stableRelease.version,
-      stableRelease.date,
-      'current',
-      stableRelease.releaseNote,
-      stableRelease.engineVersion,
-    );
-  } else {
-    result += newBrowserEntry(
-      safariBCD,
-      options.bcdBrowserName,
-      stableRelease.version,
-      'current',
-      'WebKit',
-      stableRelease.date,
-      stableRelease.releaseNote,
-      stableRelease.engineVersion,
-    );
+    releaseData.push(releaseDataEntry);
   }
 
   //
-  // Update beta release
+  // Find current release
   //
-  if (
-    safariBCD.browsers[options.bcdBrowserName].releases[betaRelease.version]
-  ) {
-    result += updateBrowserEntry(
-      safariBCD,
-      options.bcdBrowserName,
-      betaRelease.version,
-      '',
-      'beta',
-      '',
-      '',
-    );
-  } else {
-    result += newBrowserEntry(
-      safariBCD,
-      options.bcdBrowserName,
-      betaRelease.version,
-      'beta',
-      'WebKit',
-      '',
-      stableRelease.releaseNote,
-      '',
-    );
-  }
+  const dates = [];
+  releaseData.forEach((release) => {
+    if (release.channel !== 'beta') {
+      dates.push(release.date);
+    }
+  });
+  const currentDate = dates.sort().pop();
+  releaseData.forEach((release) => {
+    if (release.date === currentDate) {
+      release.channel = 'current';
+    }
+  });
 
   //
-  // Replace all old entries with 'retired'
+  // Update from releaseData object to BCD
   //
-  Object.entries(safariBCD.browsers[options.bcdBrowserName].releases).forEach(
-    ([key, entry]) => {
-      if (parseFloat(key) < stableRelease) {
+  releaseData.forEach((release) => {
+    if (!options.skippedReleases.includes(release.version)) {
+      if (
+        safariBCD.browsers[options.bcdBrowserName].releases[release.version]
+      ) {
         result += updateBrowserEntry(
```

Comment:
To prevent errors from being added to `safari.json`, I think we shouldn't update old releases – at least before #21338 land.

So I think we should add an extra test:
```suggestion
        if (release.version >= 17) {
          result += updateBrowserEntry(
```
@elchi3, what do you think?

Reply:
https://github.com/mdn/browser-compat-data/pull/21338 was merged, so this is not a problem anymore.